Index: Modules/Course/classes/class.ilObjCourseGUI.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Course/classes/class.ilObjCourseGUI.php	(revision 3262b464dd37023a8e0561fb4fe59f417eec1056)
+++ Modules/Course/classes/class.ilObjCourseGUI.php	(revision )
@@ -390,8 +390,22 @@
 			foreach ($emails as $email) {
 				$email = trim($email);
 				$etpl = new ilTemplate("tpl.crs_contact_email.html", true, true , 'Modules/Course');
-                $etpl->setVariable("EMAIL_LINK", ilMailFormCall::getLinkTarget($info, 'showSummary', array(),
-					array('type' => 'new', 'rcp_to' => $email,'sig' => $this->createMailSignature())));
+				$etpl->setVariable(
+					"EMAIL_LINK",
+					ilMailFormCall::getLinkTarget(
+						$info, 'showSummary', array(),
+						array(
+							'type'   => 'new',
+							'rcp_to' => $email,
+							'sig' => $this->createMailSignature()
+						),
+						array(
+							ilMailFormCall::CONTEXT_KEY => 'crs_context_manual',
+							'ref_id' => $this->object->getRefId(),
+							'ts'     => time()
+						)
+					)
+				);
 				$etpl->setVariable("CONTACT_EMAIL", $email);				
 				$mailString .= $etpl->get()."<br />";
 			}
Index: Modules/Course/module.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Course/module.xml	(revision 3262b464dd37023a8e0561fb4fe59f417eec1056)
+++ Modules/Course/module.xml	(revision )
@@ -37,4 +37,7 @@
 	<copage>
 		<pageobject parent_type="lobj" class_name="ilLOPage" directory="classes/Objectives"/>	
 	</copage>
+	<mailtemplates>
+		<context id="crs_context_manual" class="ilCourseMailTemplateContext" />
+	</mailtemplates>
 </module>
Index: Modules/Course/classes/class.ilCourseMailTemplateContext.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Course/classes/class.ilCourseMailTemplateContext.php	(revision )
+++ Modules/Course/classes/class.ilCourseMailTemplateContext.php	(revision )
@@ -0,0 +1,72 @@
+<?php
+
+class ilCourseMailTemplateContext extends ilMailTemplateContext
+{
+	/**
+	 * @return string
+	 */
+	public function getId()
+	{
+		return 'crs_context_manual';
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getTitle()
+	{
+		return 'Course';
+	}
+
+	/**
+	 * Return an array of placeholders
+	 * @return array
+	 */
+	public function getSpecificPlaceholders()
+	{
+		/**
+		 * @var $lng ilLanguage
+		 */
+		global $lng;
+
+		$lng->loadLanguageModule('crs');
+
+		return array(
+			'crs_title' => array(
+				'placeholder' => 'CRS_TITLE',
+				'label'       => $lng->txt('crs_title')
+			),
+			'crs_link'  => array(
+				'placeholder' => 'CRS_LINK',
+				'label'       => $lng->txt('crs_mail_permanent_link')
+			)
+		);
+	}
+
+	/**
+	 * @param string    $placeholder_id
+	 * @param array     $context_parameters
+	 * @param ilObjUser $recipient
+	 * @param bool      $html_markup
+	 * @return string
+	 */
+	public function resolveSpecificPlaceholder($placeholder_id, array $context_parameters, ilObjUser $recipient, $html_markup = false)
+	{
+		/**
+		 * @var $ilObjDataCache ilObjectDataCache
+		 */
+		global $ilObjDataCache;
+
+		if('crs_title' == $placeholder_id)
+		{
+			return $ilObjDataCache->lookupTitle($ilObjDataCache->lookupObjId($context_parameters['ref_id']));
+		}
+		else if('crs_link' == $placeholder_id)
+		{
+			require_once './Services/Link/classes/class.ilLink.php';
+			return ilLink::_getLink($context_parameters['ref_id'], 'crs');
+		}
+
+		return '';
+	}
+}
\ No newline at end of file
